### PR TITLE
Add llama3.1 405B model definition and builder

### PR DESCRIPTION
**Pull request by lessw2020**
This PR:
1 - adds the model definition and base model builder for llama3_1 405B.  
2 - Have verified it's ability to produce a 405B model via import and running within torchchat/distributed:

Screenshot 2024-08-12 at 7 23 40 PM

3 - This PR is effectively a subset (main model only, rather than QLora/Lora) of https://github.com/pytorch/torchtune/pull/1232

#### Context
What is the purpose of this PR? Is it to
- [X ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](../CONTRIBUTING.md) for some guidance on contributing.)

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](../tests/torchtune) for any new functionality
- [ ] update [docstrings](../docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

File: torchtune/models/llama3/_model_builders.py [hunk 1]
```diff
@@ -62,7 +62,25 @@ def llama3_70b() -> TransformerDecoder:
         rope_base=500000.0,
     )
 
-
+def llama3_1_405b() -> TransformerDecoder:
```
Comment (joecummings) [inline torchtune/models/llama3/_model_builders.py:65]: I think this should be removed - this is in the Llama3 folder, not 3.1

**Review by joecummings — COMMENTED** [review]
Just one comment! Otherwise looks great

**Review by felipemello1 — COMMENTED** [review]
Maybe this was intentionally left out, but shouldnt we also add a config in torchtune/configs/llama3_1/405B_full.yaml, 405B_lora.yaml?

If so, they would also need to be registered here https://github.com/pytorch/torchtune/blob/main/torchtune/_recipe_registry.py

**Comment by joecummings** [comment]
> Maybe this was intentionally left out, but shouldnt we also add a config in torchtune/configs/llama3_1/405B_full.yaml, 405B_lora.yaml?
> 
> If so, they would also need to be registered here [`main`/torchtune/_recipe_registry.py](https://github.com/pytorch/torchtune/blob/main/torchtune/_recipe_registry.py?rgh-link-date=2024-08-13T13%3A31%3A22Z)

Nope - we're just getting the model builder in here for now so that torchchat can use it.

File: torchtune/models/llama3/_model_builders.py (hunk 1 above)
Comment (lessw2020) [reply torchtune/models/llama3/_model_builders.py]: great catch - this was inadvertent.  Removing!

File: torchtune/models/llama3/_model_builders.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def llama3_70b() -> TransformerDecoder:
         rope_base=500000.0,
     )
 
-
```
Comment (joecummings) [inline torchtune/models/llama3/_model_builders.py:65]: 🙃

**Review by joecummings — APPROVED** [review]
Looks great! Will merge when CI is green.

File: torchtune/models/llama3/_model_builders.py (hunk 2 above)
Comment (lessw2020) [reply torchtune/models/llama3/_model_builders.py:65]: lol - after removing the inadvertent 405B builder, I was trying to keep the same formatting of two blank lines between each model_builder.

**Pull request merged**